hazelcast-sql/src/main/java/com/hazelcast/sql/impl/type/QueryDataType.java

```diff
@@ -148,6 +177,30 @@ public TypeKind getObjectTypeKind() {
         return objectTypeKind;
     }
 
+    public void addField(String name, QueryDataType type) {
+        if (objectFieldsBuilder == null) {
+            objectFieldsBuilder = ImmutableList.builder();
+        }
+        objectFieldsBuilder.add(new QueryDataTypeField(name, type));
+    }
+
+    public void finalizeFields() {
```

Comment:
if the method was unable to add or finalize a list, then it is worth reporting this somehow - throwing an exception or returning true\false

Reply:
[`8818039c`..`217f6dc0`][1]

[1]: https://github.com/hazelcast/hazelcast/compare/8818039c4cdcb25814b2d161a97c8032d229f536..217f6dc09634511c35cda5fc73dba20851abd051